include/pybind11/detail/smart_holder_type_casters.h

```diff
@@ -625,6 +625,7 @@ struct smart_holder_type_caster<std::shared_ptr<T>> : smart_holder_type_caster_l
     static handle cast(const std::shared_ptr<T> &src, return_value_policy policy, handle parent) {
         if (policy != return_value_policy::automatic
             && policy != return_value_policy::reference_internal
+            && policy != return_value_policy::copy
```

Comment:
Why don't you need `move` as well??

Reply:
Indeed that is needed, should be added now.